src/main/flush.ts

```diff
@@ -0,0 +1,166 @@
+import fs from 'fs';
+
+export enum FilterTypeEnum {
+  PEAK = 'PK', // ["PK",True,True]
+  LPQ = 'LPQ', // ["LPQ",False,True]
+  HPQ = 'HPQ', // ["HPQ",False,True]
+  BP = 'BP', // ["BP",False,True]
+  LS = 'LS', // ["LS",True,False]
+  HS = 'HS', // ["HS",True,False]
+  NO = 'NO', // ["NO",False,True]
+  AP = 'AP', // ["AP",False,True]
+  LSC = 'LSC', // ["LSC",True,True]
+  HSC = 'HSC', // ["HSC",True,True]
+  BWLP = 'BWLP', // ["BWLP",False,True]
+  BWHP = 'BWHP', // ["BWHP",False,True]
+  LRLP = 'LRLP', // ["LRLP",False,True]
+  LRHP = 'LRHP', // ["LRHP",False,True]
+  LSCQ = 'LSCQ', // ["LSCQ",True,True]
+  HSCQ = 'HSCQ', // ["HSCQ",True,True]
+}
+
+interface IFilter {
+  frequency: number;
+  gain: number;
+  type: FilterTypeEnum;
+  quality: number;
+}
+
+export interface IState {
+  isEnabled: boolean;
+  preAmp: number;
+  filters: IFilter[];
+}
+
+const FIXED_FREQUENCIES = [
+  32, 64, 125, 250, 500, 1000, 2000, 4000, 8000, 16000,
+];
+
+export const DEFAULT_FILTER: IFilter = {
+  frequency: 1000,
+  gain: 0,
+  quality: 1,
+  type: FilterTypeEnum.PEAK,
+};
+
+const DEFAULT_FILTERS: IFilter[] = FIXED_FREQUENCIES.map((f) => {
+  return {
+    frequency: f,
+    gain: 0,
+    quality: 1,
+    type: FilterTypeEnum.PEAK,
+  };
+});
+
+const DEFAULT_STATE: IState = {
+  isEnabled: true,
+  preAmp: 0,
+  filters: DEFAULT_FILTERS,
+};
+
+export const stateToString = (state: IState) => {
+  if (!state.isEnabled) {
+    return '';
+  }
+
+  let output: string[] = [];
+
+  output.push('Device: all');
+  output.push('Channel: all');
+
+  if (state.preAmp) {
+    output.push(`Preamp: ${state.preAmp}dB`);
+  }
+
+  // Using individual filter bands
+  output = output.concat(
+    state.filters
+      .flatMap(({ frequency, gain, type, quality }, index) => {
+        if (!gain) {
+          return '';
+        }
+        return `Filter${index}: ON ${type} Fc ${frequency} Hz Gain ${gain} dB Q ${quality}`;
+      })
+      .filter((str) => str.length > 0)
+  );
+
+  // TODO: Figure out when to use GraphicEQ
+  // `${Frequency} ${Gain};`
```

Comment:
isn't graphic eq just an alternative way of doing parameterized eq?

Reply:
yeah but i don't think it directly allows control over quality and other params...? I only left it here because Peace has code for using it instead of Filters but I haven't had the time to look over WHY they use it yet.